apps/blues-wireless/src/note_c_hooks.c

```diff
@@ -0,0 +1,154 @@
+/**
+ #
+ # Licensed to the Apache Software Foundation (ASF) under one
+ # or more contributor license agreements.  See the NOTICE file
+ # distributed with this work for additional information
+ # regarding copyright ownership.  The ASF licenses this file
+ # to you under the Apache License, Version 2.0 (the
+ # "License"); you may not use this file except in compliance
+ # with the License.  You may obtain a copy of the License at
+ #
+ #  http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing,
+ # software distributed under the License is distributed on an
+ # "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ # KIND, either express or implied.  See the License for the
+ # specific language governing permissions and limitations
+ # under the License.
+ #
+ */
+
+
+#include "note_c_hooks.h"
+#include "bsp/bsp.h"
+
+#include "hal/hal_i2c.h"
+
+#include "os/os_time.h"
+
+#include <console/console.h>
+
+static const size_t REQUEST_HEADER_SIZE = 2;
+
+const uint8_t i2c_num = 0;
+bool i2c_initialized = false;
+
+
+uint32_t
+platform_millis(void)
+{
+    return (uint32_t)(os_get_uptime_usec()/1000);
+}
+
+void
+platform_delay(uint32_t ms)
+{
+    os_time_delay((OS_TICKS_PER_SEC/1000) * ms);
+}
+
+const char *
+note_i2c_receive(uint16_t device_address, uint8_t *buffer,
+                 uint16_t size, uint32_t *available)
+{
+    {
+
+        uint8_t size_buf[2];
+        size_buf[0] = 0;
+        size_buf[1] = (uint8_t)size;
+
+        struct hal_i2c_master_data data = {
+            .address = device_address,
+            .len = sizeof(size_buf),
+            .buffer = size_buf,
+        };
+        uint8_t write_result = hal_i2c_master_write(i2c_num, &data,
+                                                OS_TICKS_PER_SEC / 10, 0);
```

Comment:
In mynewt  `OS_TICK_PER_SEC / 20`  from this like usually goes to the same column as i2c_num